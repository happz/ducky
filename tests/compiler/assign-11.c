struct foo {
  int i;
  int j;
  char c;
  int k;
  int l;
};

struct bar {
  struct foo f1;
  struct foo *f2;
};

void fn()
{
  struct foo f, *g;
  struct bar b;
  int j;

  j = f.i;
  j = g->i;
  j = g->j;
}
