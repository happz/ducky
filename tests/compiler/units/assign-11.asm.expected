
  .include "defs.asm"
  .section .rodata


.L1_global:


  .text
  .global fn

fn:
  sub sp, 26
  push r7
  push r8
  push r9
  push r10
  push r11
  push r12


  lw r12, fp[-10]
  mov r11, r12
  lw r10, fp[-12]
  add r10, 0
  lw r9, r10
  mov r11, r9
  lw r8, fp[-12]
  add r8, 2
  lw r7, r8
  mov r11, r7

.L2_fn_return:
  pop r12
  pop r11
  pop r10
  pop r9
  pop r8
  pop r7
  add sp, 26
  ret


_start:
  call &main
  int $INT_HALT
  .global _start
